lib/private/DateTimeFormatter.php

```diff
@@ -125,7 +125,7 @@ public function formatDate($timestamp, $format = 'long', \DateTimeZone $timeZone
 	 * @return string Formatted relative date string
 	 */
 	public function formatDateRelativeDay($timestamp, $format = 'long', \DateTimeZone $timeZone = null, \OCP\IL10N $l = null) {
-		if (substr($format, -1) !== '*' && substr($format, -1) !== '*') {
+		if (!str_ends_with($format, '*')) {
```

Comment:
```suggestion
		if (!str_ends_with($formatDate, '^') && !str_ends_with($formatDate, '*')) {
```

I think the duplication was due to a bad copy-pasting. Having `*` or `^` at the end of the format seem on purpose and make sense if you look at the documentation of the library that we use. Same as on line 292.

https://punic.readthedocs.io/en/3.4.0/_static/api/class-Punic.Calendar.html
![Screenshot from 2023-08-17 19-38-51](https://github.com/nextcloud/server/assets/6653109/372166fb-f100-49b5-8e27-71b61cf3968c)